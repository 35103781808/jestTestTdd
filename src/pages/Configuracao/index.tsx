import React from 'react';
import Formulario from '../../components/Formulario';
import ListaParticipantes from '../../components/ListaParticipantes';
import Rodape from '../../components/Rodape';

function Configuracao() {
  return (
    <>
      <Formulario />
      <ListaParticipantes />
      <Rodape />

    </>

  );
}

export default Configuracao;
