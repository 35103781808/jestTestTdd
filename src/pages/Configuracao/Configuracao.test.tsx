import { render } from '@testing-library/react';
import React from 'react';
import { RecoilRoot } from 'recoil';
import Configuracao from './index';

const mockNavegacao = jest.fn();

jest.mock('react-router-dom', () => ({ useNavigate: () => mockNavegacao }));

describe('a página de configuração', () => {
  test('Deve ser renderizada corretamente', () => {
    const { container } = render(
      <RecoilRoot>
        <Configuracao />
      </RecoilRoot>,
    );

    expect(container).toMatchSnapshot();
  });
});
