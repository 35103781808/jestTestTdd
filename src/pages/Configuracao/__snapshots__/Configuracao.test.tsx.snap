// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a página de configuração Deve ser renderizada corretamente 1`] = `
<div>
  <div
    class="sc-bczRLJ fHFuOq"
  >
    <div
      class="sc-gsnTZi lphAIs"
    >
      <div
        class="sc-dkzDqf fugkCd"
      >
        <img
          alt="sorteador"
          src="logo.png"
        />
      </div>
      <div
        class="sc-hKMtZM ilYMyl"
      >
        <img
          alt="participante.png"
          src="participante.png"
        />
      </div>
    </div>
    <div
      class="sc-eCYdqJ gsJbYy"
    >
      <div
        class="sc-jSMfEi aazvk"
      >
         Vamos Começar
      </div>
      <form>
        <input
          placeholder="Insira os nomes dos participantes"
          type="text"
          value=""
        />
        <button
          disabled=""
          type="submit"
        >
          Adicionar
        </button>
        
      </form>
    </div>
  </div>
  <ul />
  <footer>
    <button
      disabled=""
      type="button"
    >
      Iniciar Brincadeira
    </button>
  </footer>
</div>
`;
